providers/snyk/vulns.go

```diff
@@ -0,0 +1,244 @@
+	Exploit string `json:"exploit,omitempty"`
+
+	// The time at which this vulnerability was published.
+	PublicationTime string `json:"publication_time,omitempty"`
+	References      []struct {
+		// Descriptor for an external reference to the issue
+		Title string `json:"title,omitempty"`
+
+		// URL for an external reference to the issue
+		Url string `json:"url,omitempty"`
+	} `json:"references,omitempty"`
+}
+
+const (
+	Critical EffectiveSeverityLevel = "critical"
+	High     EffectiveSeverityLevel = "high"
+	Info     EffectiveSeverityLevel = "info"
+	Low      EffectiveSeverityLevel = "low"
+	Medium   EffectiveSeverityLevel = "medium"
+)
+
+// The type from enumeration of the issue’s severity level. This is usually set from the issue’s producer, but can be overridden by policies.
+type EffectiveSeverityLevel string
+
+type SnykIssuesDocument struct {
+	Data    []SnykIssueResource `json:"data,omitempty"`
+	Jsonapi JsonApi             `json:"jsonapi,omitempty"`
+	Links   *PaginatedLinks     `json:"links,omitempty"`
+	Meta    *IssuesMeta         `json:"meta,omitempty"`
+}
+
+func getVulnsForPurl(
+	purl string,
+	client *HttpRequest.Request,
+	orgID string,
+) (vulns []models.Vulnerability, err error) {
+	if err := validatePurl(purl); err != nil {
+		return nil, err
+	}
+
+	issuesURL := fmt.Sprintf("%s/orgs/%s/packages/%s/issues%s", SNYK_URL, orgID, url.QueryEscape(purl), SNYK_API_VERSION)
+
+	res, err := client.Get(issuesURL)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vulnerabilities (purl: %s): %w", purl, err)
+	}
+	defer res.Close()
+
+	body, err := res.Body()
+	if err != nil {
+		return nil, fmt.Errorf("could not read response body (purl: %s): %w", purl, err)
+	}
+
+	if res.StatusCode() != 200 {
+		return nil, fmt.Errorf("failed request while retrieving vulnerabilities (purl: %s, status: %s)", purl, res.Response().Status)
+	}
+
+	var response SnykIssuesDocument
+	if err = json.Unmarshal(body, &response); err != nil {
+		return nil, fmt.Errorf("could not parse response (purl: %s): %w", purl, err)
+	}
+
+	for _, v := range response.Data {
+		vuln := snykIssueToBomberVuln(v)
+		vulns = append(vulns, vuln)
+	}
+
+	return vulns, nil
+}
+
+func validatePurl(purl string) error {
+	if _, err := packageurl.FromString(purl); err != nil {
+		return fmt.Errorf("invalid purl: %w", err)
+	}
+	return nil
+}
+
+func snykIssueToBomberVuln(v SnykIssueResource) models.Vulnerability {
+	cvss := getCvss(v)
+	severity := strings.ToUpper(string(v.Attributes.EffectiveSeverityLevel))
+
+	if severity == "MEDIUM" {
+		severity = "MODERATE"
+	}
+
+	return models.Vulnerability{
+		ID:                 v.Id,
+		Title:              v.Attributes.Title,
+		Description:        v.Attributes.Description,
+		Severity:           severity,
+		Cwe:                getCwe(v),
+		CvssScore:          float64(cvss.Score),
+		CvssVector:         cvss.Vector,
+		Reference:          fmt.Sprintf("https://security.snyk.io/vuln/%s", v.Id),
```

Comment:
@garethr unfortunately we don't include links to security.snyk.io in the Issues response, hence the hacky construction. Maybe a new feature request for the service?